File: Java/JDI/jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/interfaces/complex/IPagination.java
package com.epam.jdi.uitests.core.interfaces.complex;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.annotations.JDIAction;
import com.epam.jdi.uitests.core.interfaces.base.IBaseElement;
import com.epam.jdi.uitests.core.interfaces.base.IComposite;

/**
 * Created by Roman_Iovlev on 7/29/2015.
 */
public interface IPagination extends IBaseElement, IComposite {
    /**
     * Choose Next page
     */
    @JDIAction
    void next();

    /**
     * Choose Previous page
     */
    @JDIAction
    void previous();

    /**
     * hoose First page
     */
    @JDIAction
    void first();

    /**
     * Choose Last page
     */
    @JDIAction
    void last();

    /**
     * @param index Specify page index
     *              Choose page by index
     */
    @JDIAction
    void selectPage(int index);
}

File: C#.Net/JDI/Core/UnitTestProject1/Interfaces/Complex/IPagination.cs
using JDI_Core.Interfaces.Base;

namespace JDI_Core.Interfaces.Complex
{
    public interface IPagination : IBaseElement, IComposite
    {
        /**
         * Choose Next page
         */
        //TODO[JDIAction]
        void Next();

        /**
         * Choose Previous page
         */
        //TODO[JDIAction]
        void Previous();

        /**
         * hoose First page
         */
        //TODO[JDIAction]
        void First();

        /**
         * Choose Last page
         */
        //TODO[JDIAction]
        void Last();

        /**
         * @param index Specify page index
         *              Choose page by index
         */
        //TODO[JDIAction]
        void SelectPage(int index);
    }
}
